### PR TITLE
Alternative Promise API

### PR DESCRIPTION
**Pull request by kothique**
What do you think?

```js
const sensor = require('node-dht-sensor');
const psensor = sensor.promises;

psensor.setMaxTries(42e42); // === sensor.setMaxTries

psensor.initialize(11, 7); // === sensor.initialize(11, 7);
psensor.initialize(11, 11);

const readout = psensor.readSync(11, 7); // === sensor.read(11, 7);

psensor.read(11, 11).then(
  ({ temperature, humidity }) => { /* ... */ },
  error => { /* ... */ }
);
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "author": "David Momenso",
   "name": "node-dht-sensor",
   "description": "Reads data from DHT sensors on Raspberry Pi",
-  "version": "0.2.1",
+  "version": "0.3.0",
```
Comment (kothique) [inline package.json]: Probably this should have been done in a different commit

**Comment by kothique** [comment]
I think this way is more elegant than just creating another function like `readPromise`, because in that case you would end up with two different methods having the same functionality:

```js
const sensor = require('node-dht-sensor');

sensor.read(bla, bla, (err, t, h) => bla);
sensor.readPromise(bla, bla).then(bla, bla);
```

And it is usually not a good idea to mix callbacks and promises.

**Comment by kothique** [comment]
I'll add test and documentation if you approve this.

**Comment by momenso** [comment]
I like the idea of having a promise API! :+1: 
This is really cool.

File: package.json (hunk 1 above)
Comment (momenso) [reply package.json]: Yes, no need to bump version in the pull-request.

File: test/node-dht-sensor.js [hunk 2]
```diff
@@ -87,6 +87,22 @@ describe('Read sensor', () => {
             });
         });
     });
+    describe('Asynchronously (promise)', () => {
+        it('should obtain temperature and humidity', async () => {
+            const { temperature, humidity } = await sensor.read(SENSOR_TYPE, GPIO_PIN);
```
Comment (momenso) [inline test/node-dht-sensor.js]: Shouldn't we be calling read from promises API?

Comment (kothique) [reply test/node-dht-sensor.js]: Oh, gosh. I'll fix it.
